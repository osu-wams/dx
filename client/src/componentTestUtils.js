import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
