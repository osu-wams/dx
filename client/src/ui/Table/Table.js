import styled from 'styled-components';
import PropTypes from 'prop-types';
import styledMap from 'styled-map';

const Table = styled.table`
  border: ${styledMap('variant', {
    basic: 'none',
    compact: 'none',
    default: '1px solid rgba(34, 36, 38, 0.15)'
  })};
  border-radius: 0.28571429rem;
  border-collapse: separate;
  border-spacing: 0;
  font-size: ${styledMap('variant', {
    compact: '1.4rem',
    default: 'inherit'
  })};
  overflow-x: scroll;
  td {
    padding: ${styledMap('variant', {
      compact: '.2rem .4rem',
      default: '0.4rem 0.8rem'
    })};
  }
  td:not(:first-child) {
    border-left: 1px solid rgba(34, 36, 38, 0.15);
  }
  tr:not(:first-child) td {
    border-top: 1px solid rgba(34, 36, 38, 0.15);
  }

  ${({ striped }) =>
    striped &&
    `
    tr:nth-child(even) {
      background: #f9fafb;
    }
  `};
`;

Table.propTypes = {
  variant: PropTypes.oneOf(['default', 'compact', 'basic']),
  striped: PropTypes.bool
};

Table.defaultProps = {
  variant: 'default',
  striped: false
};

export default Table;
