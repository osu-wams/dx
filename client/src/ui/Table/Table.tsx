/* eslint-disable no-unexpected-multiline */
import styled from 'styled-components';
import styledMap from 'styled-map';
import { Color } from '../../theme';

const Table = styled.table<{
  striped?: boolean;
  stretch?: boolean;
  variant: 'basic' | 'compact' | 'default';
}>`
  border: ${styledMap('variant', {
    basic: 'none',
    compact: 'none',
    default: '1px solid rgba(34, 36, 38, 0.15)'
  })};
  border-collapse: separate;
  border-spacing: 0;
  color: ${Color['neutral-600']};
  font-size: ${styledMap('variant', {
    compact: '1.4rem',
    default: 'inherit'
  })};
  overflow-x: scroll;
  td {
    padding: ${styledMap('variant', {
      compact: '.2rem .4rem',
      default: '0.4rem 0.8rem'
    })};
  }
  td:not(:first-child),
  th:not(:first-child) {
    border-left: ${styledMap('variant', {
      basic: 'none',
      compact: 'none',
      default: '1px solid rgba(34, 36, 38, 0.15)'
    })};
  }
  tr:not(:first-child) td {
    border-top: 1px solid rgba(34, 36, 38, 0.15);
  }
  ${({ stretch }) =>
    stretch &&
    `width: 100%;
  `};
  ${({ striped }) =>
    striped &&
    `
    tr:nth-child(even) {
      background: #f9fafb;
    }
  `};
`;

export default Table;
