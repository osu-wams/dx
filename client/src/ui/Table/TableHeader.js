import styled from 'styled-components';

const TableHeader = styled.thead``;

export default TableHeader;
