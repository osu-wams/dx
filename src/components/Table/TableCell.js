import styled from 'styled-components';

const TableCell = styled.td``;

export default TableCell;
