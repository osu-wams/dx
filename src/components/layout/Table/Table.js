import styled from 'styled-components';
import PropTypes from 'prop-types';

const Table = styled.table`
  border: 1px solid rgba(34, 36, 38, 0.15);
  border-radius: 0.28571429rem;
  border-collapse: separate;
  border-spacing: 0;
  td {
    padding: 0.4em 0.8em;
  }
  td:not(:first-child) {
    border-left: 1px solid rgba(34, 36, 38, 0.15);
  }
  tr:not(:first-child) td {
    border-top: 1px solid rgba(34, 36, 38, 0.15);
  }

  ${({ variant }) => {
    switch (variant) {
      case 'basic':
        return `
          border: none;
        `;

      default:
        return null;
    }
  }};

  ${({ striped }) =>
    striped &&
    `
    tr:nth-child(even) {
      background: #f9fafb;
    }
  `};
`;

Table.propTypes = {
  variant: PropTypes.oneOf(['default', 'basic']),
  striped: PropTypes.bool
};

Table.defaultProps = {
  variant: 'default',
  striped: false
};

export default Table;
