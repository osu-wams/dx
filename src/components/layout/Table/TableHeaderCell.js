import styled from 'styled-components';

const TableHeaderCell = styled.th`
  font-weight: bold;
  border-bottom: 1px solid rgba(34, 36, 38, 0.15);
  :not(:first-child) {
    border-left: 1px solid rgba(34, 36, 38, 0.15);
  }
`;

export default TableHeaderCell;
