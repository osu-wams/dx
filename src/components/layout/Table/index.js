import Table from './Table';
import TableBody from './TableBody';
import TableRow from './TableRow';
import TableCell from './TableCell';
import TableHeader from './TableHeader';
import TableHeaderCell from './TableHeaderCell';

export { Table, TableBody, TableRow, TableCell, TableHeader, TableHeaderCell };
