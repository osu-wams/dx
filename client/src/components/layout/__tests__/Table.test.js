import React from 'react';
import { render } from '../../../componentTestUtils';
import { Table, TableBody, TableRow, TableCell } from '../Table';

const DefaultTable = () => (
  <div>
    <Table>
      <caption>Default Table</caption>
      <TableBody>
        <TableRow>
          <TableCell>One col</TableCell>
          <TableCell>Two col</TableCell>
        </TableRow>
      </TableBody>
    </Table>
  </div>
);

const BasicTable = () => (
  <div>
    <Table variant="basic">
      <caption>Basic Table</caption>
      <tbody>
        <TableRow>
          <TableCell>One col</TableCell>
          <TableCell>Two col</TableCell>
        </TableRow>
      </tbody>
    </Table>
  </div>
);

const StripedTable = () => (
  <div>
    <Table variant="basic" striped>
      <caption>Basic Table</caption>
      <tbody>
        <TableRow>
          <TableCell>One col</TableCell>
          <TableCell>Two col</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>One col</TableCell>
          <TableCell>Two col</TableCell>
        </TableRow>
      </tbody>
    </Table>
  </div>
);

test('DefaultTable gets rendered with border around table', () => {
  const { container } = render(<DefaultTable />);
  expect(container.firstChild).toMatchSnapshot();
});

test('BasicTable no border around table element', () => {
  const { container } = render(<BasicTable />);
  expect(container.firstChild).toMatchSnapshot();
});

test('StripedTable alternates background color on rows', () => {
  const { container } = render(<StripedTable />);
  expect(container.firstChild).toMatchSnapshot();
});
