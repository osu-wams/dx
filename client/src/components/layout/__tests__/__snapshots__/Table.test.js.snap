// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTable no border around table element 1`] = `
.c0 {
  border: 1px solid rgba(34,36,38,0.15);
  border-radius: 0.28571429rem;
  border-collapse: separate;
  border-spacing: 0;
  border: none;
}

.c0 td {
  padding: 0.4em 0.8em;
}

.c0 td:not(:first-child) {
  border-left: 1px solid rgba(34,36,38,0.15);
}

.c0 tr:not(:first-child) td {
  border-top: 1px solid rgba(34,36,38,0.15);
}

<div>
  <table
    class="c0"
  >
    <caption>
      Basic Table
    </caption>
    <tbody>
      <tr
        class=""
      >
        <td
          class=""
        >
          One col
        </td>
        <td
          class=""
        >
          Two col
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DefaultTable gets rendered with border around table 1`] = `
.c0 {
  border: 1px solid rgba(34,36,38,0.15);
  border-radius: 0.28571429rem;
  border-collapse: separate;
  border-spacing: 0;
}

.c0 td {
  padding: 0.4em 0.8em;
}

.c0 td:not(:first-child) {
  border-left: 1px solid rgba(34,36,38,0.15);
}

.c0 tr:not(:first-child) td {
  border-top: 1px solid rgba(34,36,38,0.15);
}

<div>
  <table
    class="c0"
  >
    <caption>
      Default Table
    </caption>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class=""
        >
          One col
        </td>
        <td
          class=""
        >
          Two col
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`StripedTable alternates background color on rows 1`] = `
.c0 {
  border: 1px solid rgba(34,36,38,0.15);
  border-radius: 0.28571429rem;
  border-collapse: separate;
  border-spacing: 0;
  border: none;
}

.c0 td {
  padding: 0.4em 0.8em;
}

.c0 td:not(:first-child) {
  border-left: 1px solid rgba(34,36,38,0.15);
}

.c0 tr:not(:first-child) td {
  border-top: 1px solid rgba(34,36,38,0.15);
}

.c0 tr:nth-child(even) {
  background: #f9fafb;
}

<div>
  <table
    class="c0"
  >
    <caption>
      Basic Table
    </caption>
    <tbody>
      <tr
        class=""
      >
        <td
          class=""
        >
          One col
        </td>
        <td
          class=""
        >
          Two col
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class=""
        >
          One col
        </td>
        <td
          class=""
        >
          Two col
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
